[CS2103-F10-1] NurseTraverse

crazoter opened the pull request:
@francislow @sandydays @cheongsiuhong @gabrielchao

damithc commented:
PR name corrected.

JermyTan commented on docs/DeveloperGuide.adoc:
```diff
@@ -334,26 +434,81 @@ Use case ends.
+*MSS*
+
+1.  User requests to open note GUI for a specific patient
+2.  User edit the patient’s notes
```
Perhaps the response of the system could be more explicit, eg "2. Application opens note GUI for specified patient".

From CS2103T-F11-2 tutorial

JermyTan commented on docs/DeveloperGuide.adoc:
```diff
@@ -334,26 +434,81 @@ Use case ends.
+*Extensions*
+
+[none]
+* 1a. Invalid command
```
Perhaps this could be more specific. Do you mean invalid input format or this command should not work for specific types of patient?

JermyTan commented on docs/DeveloperGuide.adoc:
```diff
@@ -334,26 +434,81 @@ Use case ends.
+
+1.  User requests to list persons
+2.  Application shows a list of persons
+3.  User picks one patient from the list and add appointment date and time through command line
```
The phrase "through command line" could be classified as internal details which could be unnecessary for use case.

Wingedevil commented:
Referencing https://ay1920s1-cs2103-f10-1.github.io/main/images/DeleteSequenceDiagram.png in the Design section is a very low level diagram, perhaps consider moving it into the Implementation section and referencing the relevant subsections.

Storage component could use more descriptions on the design used.

Referencing https://ay1920s1-cs2103-f10-1.github.io/main/images/ExportSequenceDiagram.png,
`getCsvStringFromPersons(personList)` self-invocation should have another activation block.
`writeToFIle(name, csvString)` should be a solid instead of a dotted line.

Throughout the whole developer guide, please take note to be consistent when using dashes, be it en-dash or em-dash. 

🥔🥔🥔Potato🥔🥔🥔

francislow committed:
Merge pull request #188 from francislow/dry-run-pe-bug-fixes

Dry run pe fixes

gabrielchao committed:
Merge pull request #14 from AY1920S1-CS2103-F10-1/master

Merge team repo

crazoter committed:
Fix Visit display bugs

gabrielchao committed:
Update DG UI and Model design considerations

crazoter committed:
Update UG to include tab information

gabrielchao committed:
Merge branch 'master' into undo-redo-v1_4

gabrielchao committed:
Add newline to end of file

gabrielchao committed:
Merge branch 'undo-redo-v1_4' of https://github.com/gabrielchao/main into undo-redo-v1_4

gabrielchao committed:
Merge pull request #191 from gabrielchao/undo-redo-v1_4

Update DG

crazoter committed:
Fix single index error messages

crazoter committed:
Ensure ongoing visit error message comes first

crazoter committed:
Fix checkstyle

crazoter committed:
Merge branch 'master' into branch-visit-patch

gabrielchao committed:
Add deep copy infobox

gabrielchao committed:
Merge branch 'master' into undo-redo-v1_4

gabrielchao committed:
Merge pull request #192 from gabrielchao/undo-redo-v1_4

Update DG

cheongsiuhong committed:
fix ug bugs for import export.
throw error for empty files in import.
phone numbers now limited to 8 digits.
update error for phone numbers that are too long.

cheongsiuhong committed:
f checkstyle

cheongsiuhong committed:
fix phone numbers test case

cheongsiuhong committed:
Merge pull request #194 from cheongsiuhong/branch_bug_fixing

import export + misc bug fixing

crazoter committed:
Refactor parseCommand

sandydays committed:
Add Appointment Book to storage and modify tests.

sandydays committed:
Add appointment book to history.

cheongsiuhong committed:
add csvUtil tests
add export command tests
add sample csv file in ./imports
refactor import, export and csvUtil to manage paths differently.

cheongsiuhong committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-1/main into branch-import-export-testing

cheongsiuhong committed:
add import merge command test.
phone numbers now up to 50 digits.

cheongsiuhong committed:
add import replace command test.
revert phone limit to 8 digits.

gabrielchao committed:
Add RedoCommandTest, LogicManagerIntegrationTest

gabrielchao committed:
Add more RedoCommand and HistoryManager tests

gabrielchao committed:
Merge branch 'master' into undo-redo-v1_4

gabrielchao committed:
Merge pull request #15 from AY1920S1-CS2103-F10-1/master

Merge team repo

gabrielchao committed:
Add case sensitive Name.isSameName(), UG patient identity infobox

gabrielchao committed:
Add more sections to UndoRedoHistory.adoc

francislow committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-1/main into branch-autocomplete-v1.4

francislow committed:
Change select key to RIGHT

gabrielchao committed:
Merge pull request #197 from gabrielchao/patient_identity

Refine patient identity definition

gabrielchao committed:
Merge branch 'master' into undo-redo-v1_4

francislow committed:
Add newline at eof

gabrielchao committed:
Add newline to end of file

gabrielchao committed:
Merge branch 'undo-redo-v1_4' of https://github.com/gabrielchao/main into undo-redo-v1_4

gabrielchao committed:
Merge pull request #196 from gabrielchao/undo-redo-v1_4

Add tests

cheongsiuhong committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-1/main into branch-import-export-testing

gabrielchao committed:
Refactor name of PatientBookParser to AppParser

francislow committed:
Change class name

francislow committed:
Update ug

gabrielchao committed:
Commit unstaged changes

gabrielchao committed:
Merge branch 'master' into undo-redo-v1_4

francislow committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-1/main into branch-autocomplete-v1.4

crazoter committed:
Revert "Refactor parseCommand"

This reverts commit cdc5fc85cd356d24cfd487ace2f0acc176c6d035.

crazoter committed:
Revert "Fix checkstyle"

This reverts commit b7043776a4ff5d80a53eeec76a26eb8033977acf.

crazoter committed:
Revert "Ensure ongoing visit error message comes first"

This reverts commit b7a8fe67de45e74c9b6e6f6bf64a577fa513a267.

sandydays committed:
Fix bugs with changes to staged appointment book.

sandydays committed:
Ensure correct behaviour for edit and delete pat commands due to appointment dependence on patients.

cheongsiuhong committed:
add export command parser tests.

crazoter committed:
Merge branch 'master' into branch-visit-patch

crazoter committed:
Fix padding for VisitTask

crazoter committed:
Merge branch 'branch-visit-patch' of https://github.com/crazoter/main into branch-visit-patch

cheongsiuhong committed:
add import command parser tests.
add file name validation.

cheongsiuhong committed:
update info message for a few commands.
update sample util data.
update tag to allow spaces.

cheongsiuhong committed:
update import export user guide

crazoter committed:
Update DG

gabrielchao committed:
Merge pull request #198 from gabrielchao/undo-redo-v1_4

Refactor PatientBookParser to AppParser

cheongsiuhong committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-1/main into branch-import-export-testing

crazoter committed:
Merge branch 'master' into branch-visit-patch

crazoter committed:
Update documentation

crazoter committed:
Merge branch 'master' into branch-visit-documentation

cheongsiuhong committed:
update ug for readability

cheongsiuhong committed:
update ug csv example.

cheongsiuhong committed:
delete outdated example

crazoter committed:
Update PPP

crazoter committed:
Merge branch 'branch-visit-documentation' of https://github.com/crazoter/main into branch-visit-documentation

cheongsiuhong committed:
some dg changes

cheongsiuhong committed:
fix export command equals

cheongsiuhong committed:
add seq diagrams

cheongsiuhong committed:
remove unused import

sandydays committed:
Do not allow clashing appoointments.

sandydays committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-1/main into testing_appt

sandydays committed:
Merge from master, fix tests, add baseline appointments that have already finished.

cheongsiuhong committed:
Merge pull request #195 from cheongsiuhong/branch-import-export-testing

Import Export testing

cheongsiuhong committed:
fix ug links

cheongsiuhong committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-1/main into branch_dg_ppp

 Conflicts:
	docs/userguide/ImportExport.adoc

cheongsiuhong committed:
edit siuhong PPP.
purge learning outcomes and other unneeded adocs.
sanitize dev guide section 2.1.
sanitize dev guide Setting Up section.

cheongsiuhong committed:
added an accidentally deleted file(UsingCheckstyle.adoc)

cheongsiuhong committed:
sigh

cheongsiuhong committed:
update ppp, dg and ug.

crazoter committed:
Merge branch 'master' into branch-visit-documentation

cheongsiuhong committed:
some sanitizing

crazoter committed:
Merge pull request #199 from crazoter/branch-visit-documentation

Update documentation

crazoter committed:
Merge branch 'master' into branch-visit-patch

cheongsiuhong committed:
update ui sample pic

cheongsiuhong committed:
update diagrams

cheongsiuhong committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-1/main into branch_dg_ppp

cheongsiuhong committed:
update diagram widths

cheongsiuhong committed:
correct diagrams

sandydays committed:
Improve UI of appointments.

sandydays committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-1/main into testing_appt

crazoter committed:
Remove trailing whitespace

sandydays committed:
Fix checkstyle errors.

sandydays committed:
Fix more checkstyle issues.

sandydays committed:
Minor fix of checkstyle.

cheongsiuhong committed:
Merge pull request #200 from cheongsiuhong/branch_dg_ppp

Final documentation updates.

crazoter committed:
Fix test case

crazoter committed:
Merge branch 'master' into branch-visit-patch

crazoter committed:
Merge pull request #190 from crazoter/branch-visit-patch

Fix Visit display bugs

crazoter committed:
Merge branch 'master' into branch-autocomplete-v1.4

sandydays committed:
Update some tests.

sandydays committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-1/main into testing_appt

sandydays committed:
Fix some formatting errors.

sandydays committed:
Fix style errors

sandydays committed:
Fix error.

sandydays committed:
Fix codacy issues.

gabrielchao committed:
Merge pull request #16 from AY1920S1-CS2103-F10-1/master

Merge team repo

gabrielchao committed:
Merge pull request #17 from gabrielchao/master

Merge team repo

francislow committed:
Merge pull request #189 from francislow/branch-autocomplete-v1.4

Autocomplete v1.4 increments

gabrielchao committed:
Update PPP

gabrielchao committed:
Merge branch 'master' into undo-redo-v1_4

gabrielchao committed:
Merge pull request #204 from gabrielchao/undo-redo-v1_4

Update PPP

francislow committed:
Draft PPP

francislow committed:
Update PPP

francislow committed:
Merge pull request #205 from francislow/branch-autocomplete-v1.4

Initial PPP

sandydays committed:
Fix merge conflicts, add appt-delete-permanent command.

sandydays committed:
Minor javadoc comment fix.

sandydays committed:
Fix edit appt command bug.

sandydays committed:
Fix checkstyle.

sandydays committed:
Refactoring.

gabrielchao committed:
Add RepoSense annotations

gabrielchao committed:
Merge branch 'master' into undo-redo-v1_4

francislow committed:
Add index suggestion for app-undo command

francislow committed:
Abstract out logic in autocomplete ui to a manager class

crazoter committed:
Merge pull request #201 from sandydays/testing_appt

Appointment Increments

gabrielchao committed:
Merge branch 'master' into undo-redo-v1_4

gabrielchao committed:
Merge pull request #206 from gabrielchao/undo-redo-v1_4

Add RepoSense annotations

gabrielchao committed:
Add more tests

gabrielchao committed:
Merge branch 'master' into undo-redo-v1_4

gabrielchao committed:
Add HistoryRecordTest

gabrielchao committed:
Add UndoCommandParserTest

gabrielchao committed:
Add a UndoCommandParser test

francislow committed:
Add sequence diagram

crazoter committed:
Update release version, Fix exit in UG

francislow committed:
Update dg

crazoter committed:
Update UG

crazoter committed:
Fix UndoRedo UG

crazoter committed:
Fix incomplete sentence.

francislow committed:
update dg

francislow committed:
Update dg 2

crazoter committed:
Update DG

crazoter committed:
Update PPP

crazoter committed:
Update PPP

gabrielchao committed:
Merge pull request #207 from gabrielchao/undo-redo-v1_4

Add tests

crazoter committed:
Ensure pat-clear fails if ongoing visit

crazoter committed:
Merge branch 'master' into branch-matt-tests

crazoter committed:
Update Storage class diagram

crazoter committed:
Merge branch 'branch-matt-tests' of https://github.com/crazoter/main into branch-matt-tests

francislow committed:
Update dg with diagrams

gabrielchao committed:
Update DG Model section

crazoter committed:
Update Logic portion of DG

crazoter committed:
Merge pull request #208 from crazoter/branch-matt-tests

Fix UG, DG, PPP

crazoter committed:
Update Appendices

crazoter committed:
Merge branch 'master' into branch-matt-tests

sandydays committed:
Update UG, fix issues with recurring appointments and clashing when reading from storage.

sandydays committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-1/main into testing_appt

gabrielchao committed:
Fix EditCommand same identity bug and add tests for it

sandydays committed:
Fix checkstyle errors.

gabrielchao committed:
Merge branch 'master' into undo-redo-v1_4

gabrielchao committed:
Export ModelClassDiagram.png at higher resolution

gabrielchao committed:
Merge branch 'undo-redo-v1_4' of https://github.com/gabrielchao/main into undo-redo-v1_4

francislow committed:
Refine ug dg

francislow committed:
Code clean up 1
Add autocomplete panel manager class

gabrielchao committed:
Fix typo

gabrielchao committed:
Update DG

crazoter committed:
Merge pull request #210 from crazoter/branch-matt-tests

Update Appendices

crazoter committed:
Merge branch 'master' into undo-redo-v1_4

crazoter committed:
Merge pull request #212 from gabrielchao/undo-redo-v1_4

Fix EditCommand bug, update docs

francislow committed:
Refactoring and fix checkstyle error

francislow committed:
Fix checkstyle

francislow committed:
Update ui section for dg

francislow committed:
Update glossary section of dg

francislow committed:
Add eof newline

crazoter committed:
Update PPP

francislow committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-1/main into branch-autocomplete-v1.4

# Conflicts:
#	docs/DeveloperGuide.adoc

francislow committed:
Add new appt commands to storage

francislow committed:
Update new appt commands description

crazoter committed:
Update DG instructions for manual testing

crazoter committed:
Merge pull request #214 from crazoter/branch-update-ppp

Update documentation

francislow committed:
Code clean up

sandydays committed:
Update DG, PPP, and fix appt-edit errors

sandydays committed:
Checkstyle errors.

sandydays committed:
Checkstyle fix

sandydays committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-1/main into testing_appt

francislow committed:
Fix bug

sandydays committed:
Add tests.

francislow committed:
Code clean up

francislow committed:
Update test cases

sandydays committed:
Fix recurring date time printing.

francislow committed:
Fix checkstyle

cheongsiuhong committed:
Update documentation, csv file samples

francislow committed:
Fix travis error

sandydays committed:
Add testing documentation.

francislow committed:
Fix codacy issue

cheongsiuhong committed:
more documentation updates

francislow committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-1/main into branch-autocomplete-v1.4

francislow committed:
Misc update

cheongsiuhong committed:
even more documentation updates

cheongsiuhong committed:
claim some ownership because i am possessive

sandydays committed:
Merge pull request #211 from sandydays/testing_appt

Appointments v1.4 final increments

sandydays committed:
Merge branch 'master' into documentation_appt

cheongsiuhong committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-1/main into final_sprint

sandydays committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-1/main into documentation_appt

sandydays committed:
Minor fix.

sandydays committed:
Merge branch 'documentation_appt' of https://github.com/sandydays/main into documentation_appt

cheongsiuhong committed:
import replace command deletes all appointments.
final commit!!!

crazoter committed:
Update PPP to include more of UG

francislow committed:
Update PPP and dg

francislow committed:
Add @@author for reposense

crazoter committed:
Merge pull request #218 from crazoter/branch-update-ppp2

Update PPP to include more of UG

cheongsiuhong committed:
final commit for real?

sandydays committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-1/main into documentation_appt

francislow committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-1/main into branch-autocomplete-v1.4

francislow committed:
Merge pull request #213 from francislow/branch-autocomplete-v1.4

Autocomplete tweaks

sandydays committed:
Change as requested.

cheongsiuhong committed:
csv file fixes

sandydays committed:
Labelling.

sandydays committed:
Merge branch 'master' into documentation_appt

cheongsiuhong committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-1/main into final_sprint

# Conflicts:
#	docs/DeveloperGuide.adoc

cheongsiuhong committed:
fixed a boo boo

francislow committed:
Merge pull request #217 from sandydays/documentation_appt

Add testing documentation for appointments

cheongsiuhong committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-1/main into final_sprint

crazoter committed:
Remove colons

cheongsiuhong committed:
Merge pull request #216 from cheongsiuhong/final_sprint

Final Import Export updates

crazoter committed:
Merge branch 'master' into branch-update-dg1

gabrielchao committed:
Merge pull request #18 from AY1920S1-CS2103-F10-1/master

Merge team repo

crazoter committed:
Merge pull request #219 from crazoter/branch-update-dg1

Remove colons for DG

gabrielchao committed:
Update sequence diagrams .puml

gabrielchao committed:
Update DG

gabrielchao committed:
Update DG testing appendix

gabrielchao committed:
Merge branch 'master' into final_changes

gabrielchao committed:
Update PPP

gabrielchao committed:
Merge branch 'final_changes' of https://github.com/gabrielchao/main into final_changes

gabrielchao committed:
Merge pull request #220 from gabrielchao/final_changes

Docs changes

gabrielchao committed:
Reduce DG image heights

gabrielchao committed:
Remove double quotes from image 'height' tag

gabrielchao committed:
Change image 'height' tags to 'width'

gabrielchao committed:
Adjust image dimensions, use captions instead of in-image titles

gabrielchao committed:
Remove .pdf

gabrielchao committed:
Merge branch 'master' into final_changes

gabrielchao committed:
Merge pull request #221 from gabrielchao/final_changes

Docs changes

sandydays committed:
Change PPP to 10 pages.

sandydays committed:
Merge branch 'documentation_appt' of https://github.com/sandydays/main into documentation_appt

sandydays committed:
Merge branch 'master' into documentation_appt

sandydays committed:
Merge pull request #223 from sandydays/documentation_appt

Documentation appt
